payment.yaml

```diff
@@ -0,0 +1,187 @@
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: Payment Ecosystem
+  description: Internal API that supplies abstraction over blockchain
+servers:
+  - url: 'https://api.kinmarketplace.internal/v1/'
+paths:
+  /transactions/{transaction_id}:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+      - $ref: '#/components/parameters/BlockingTime'
+      - in: path
+        name: transaction_id
+        schema:
+          type: string
+        required: true
+        description: blockchain transaction id
+    get:
+      tags:
+        - Transactions
+      description: get transaction info
+      operationId: getTransaction
+      responses:
+        '200':
+          description: Successfully returned transaction info
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Transaction'
+        '404':
+          description: Transaction not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        default:
+          description: Generic Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+                
+  /wallets/{public_address}:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+      - $ref: '#/components/parameters/BlockingTime'
+      - in: path
+        name: public_address
+        schema:
+          type: string
+        required: true
+        description: public address of wallet to create/ fund in native currency
+    post:
+      tags:
+        - Transactions
+
+      description: get transaction info
+      operationId: createWallet
+      responses:
+        '204':
+          description: Successfully created wallet
+        default:
+          description: Generic Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  
+  /wallets/{public_address}/transactions:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+      - $ref: '#/components/parameters/BlockingTime'
+      - in: path
+        name: public_address
+        schema:
+          type: string
+        required: true
+        description: public address of wallet
+    post:
+      tags:
+        - Transactions
+      description: pay a user some KIN
+      operationId: payTo
+      requestBody:
+        description: amount
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                amount:
+                  type: integer  # in microKIN
```

Comment:
better float, the sdk works in floats

Reply:
doesn't the SDK work with Decimal? I think the only way to overcome precision issues, will be to use string CC @fuzzyami @YohayBar @nitzantomer